cms/djangoapps/contentstore/views/course.py

```diff
@@ -469,6 +472,56 @@ def _get_rerun_link_for_item(course_key):
     return reverse_course_url('course_rerun_handler', course_key)
 
 
+def _ora1_deprecation_info(course_id, advanced_modules):
+    """
+    Returns information about ORA1
+        * Modules presence in the Advanced Module List.
+        * Components presence in the course outline.
+
+    Arguments:
+        course_id (CourseKey): course id
+        advanced_modules (list): advance module list
+
+    Returns:
+        Dict with following keys:
+
+        ora1_enabled (bool): True if "peergrading" and/or "combinedopenended" are in the Advanced Module List else False
+        ora1_components (list): List of ora1 components and their parent's url
+        advance_settings_url (str): URL to advance settings page
+    """
+    data = {
+        'ora1_enabled': 'peergrading' in advanced_modules or 'combinedopenended' in advanced_modules,
+        'ora1_components': [],
+        'advance_settings_url': reverse_course_url('advanced_settings_handler', course_id)
+    }
+
+    cache_key = 'ora1.components.{course}'.format(course=course_id)
+
+    ora1_components = cache.get(cache_key)  # pylint: disable=maybe-no-member
+    if not ora1_components:
+        try:
+            course_structure = CourseStructure.objects.get(course_id=course_id)
+            ordered_blocks = course_structure.ordered_blocks
+        except CourseStructure.DoesNotExist:
+            return data
+
+        store = modulestore()
+        ora1_components = []
+        for __, block in ordered_blocks.items():
+            if block['block_type'] in ['peergrading', 'combinedopenended']:
+                item = store.get_item(UsageKey.from_string(block['parent']))
+                if not getattr(item, 'is_draft', False):
+                    ora1_components.append(
+                        [reverse_usage_url('container_handler', block['parent']), block['display_name']]
+                    )
+
+        cache.set(cache_key, ora1_components, 60 * 5)  # pylint: disable=maybe-no-member
```

Comment:
Something like that for making the key, yeah. Though again, I'd like to see the querying and caching done behind the api.py abstraction. If the method in api.py you called was smart enough to accept as a parameter the block_types you were interested in, then it could just return you that data, and the code here wouldn't need to worry about cache invalidation and such -- that should be a concern of course_structures.